src/python/arcor2_arserver/helpers.py

```diff
@@ -10,3 +13,53 @@ def unique_name(name: str, existing_names: Set[str]) -> None:
 
     if name in existing_names:
         raise Arcor2Exception("Name already exists.")
+
+
+@retry(exc=CannotLock, tries=5, delay=0.5)
+@asynccontextmanager
+async def ctx_write_lock(obj_ids: Union[str, List[str]], owner: str = LOCK.SERVER_NAME) -> AsyncGenerator[None, None]:
+
+    if isinstance(obj_ids, str):
+        obj_ids = [obj_ids]
+
+    locked = []
+    for obj_id in obj_ids:
```

Comment:
Neměl bych tady mít nějaký způsobem zamknutý ten LOCK objekt a v rámci toho zámku zamknout objekty které potřebuji? Aby to zamčení jednotlivých objektů proběhlo v jedné kritické sekci. Vidím tady await... Tzn. že kdyby se odněkud zamykaly objekty A, B, C a odjinud C, B, A, tak ani na jednom místě nebude možné to zamykání dokončit, ne? Nebo se pletu? Při odemykání by pak snad problém nastat neměl.

Reply:
...možná by ta metody write_lock (která uvnitř zamčení používá) mohla brát pole ID a tenhle cyklus by se odehrál v ní. Celou dobu pod zámkem :)